script.go

```diff
@@ -858,6 +871,15 @@ func (p *Pipe) Wait() {
 	}
 }
 
+// WithEnv sets the environment for subsequent [Pipe.Exec] and [Pipe.ExecForEach] commands
+// to the string slice env. This will override the default process environment variables
+// when executing commands run via [Pipe.Exec] or [Pipe.ExecForEach]. This will not affect
+// the environment outside of [Pipe.Exec] or [Pipe.ExecForEach].
```

Comment:
"The environment outside" is a bit vague—shall we say explicitly that it doesn't affect the current _process's_ environment, in the way that `os.Setenv` would? (I can imagine people getting confused that this doesn't behave the way `Setenv` does.)